react/components/admin/StoreModal.tsx

```diff
@@ -0,0 +1,297 @@
+import type { ChangeEvent, Dispatch, FC, SetStateAction } from 'react'
+import React, { useEffect, useState } from 'react'
+import { FormattedMessage } from 'react-intl'
+import {
+  Button,
+  Input,
+  Modal,
+  EXPERIMENTAL_Select as Select,
+} from 'vtex.styleguide'
+import { v4 as createId } from 'uuid'
+
+import {
+  COUNTRY,
+  GLOVO_STORE_ID,
+  POSTAL_CODE,
+  PRIMARY,
+  SALES_CHANNEL,
+  AFFILIATE_ID,
+  STORE_NAME,
+} from '../../constants'
+import { IconGlovo } from '../../icons/IconGlovo'
+import { countries, validateInputs } from '../../utils'
+import type { AddOrEditStore } from './Stores'
+
+interface Props {
+  isOpen: boolean
+  setIsOpen: Dispatch<SetStateAction<AddOrEditStore>>
+  saving: boolean
+  store: StoreInfo | null
+  addStore: (store: StoreInfo) => Promise<void>
+  editStore: (store: StoreInfo) => Promise<void>
+}
+
+interface SelectOption {
+  label: string
+  value: string
+}
+
+const StoreModal: FC<Props> = ({
+  isOpen,
+  setIsOpen,
+  saving,
+  store,
+  addStore,
+  editStore,
+}) => {
+  const [error, setError] = useState(false)
+  const [storeInfo, setStoreInfo] = useState({} as StoreInfo)
+  const [selectedCountry, setSelectedCountry] = useState<SelectOption | null>(
+    null
+  )
+
+  useEffect(() => {
+    if (!store) {
+      setStoreInfo({
+        ...storeInfo,
+        id: createId(),
+      })
+
+      return
+    }
+
+    for (const country of countries) {
+      if (country.value === store.country) {
+        setSelectedCountry({
+          label: country.label,
+          value: country.value,
+        })
+      }
+    }
+
+    setStoreInfo(store)
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [store])
+
+  const handleCloseModal = () => {
+    setStoreInfo({} as StoreInfo)
+
+    setIsOpen({
+      isOpen: false,
+      loading: false,
+      store: null,
+    })
+  }
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const value =
+      e.target.id === AFFILIATE_ID
+        ? e.target.value.toUpperCase().replace(/[AEIOU]/, '')
+        : e.target.value
+
+    setStoreInfo({
+      ...storeInfo,
+      [e.target.id]: value,
+    })
+  }
+
+  const handleSelect = (country: { label: string; value: string }) => {
+    setStoreInfo({
+      ...storeInfo,
+      country: country.value,
+    })
+
+    setSelectedCountry(country)
+  }
+
+  const validate = () => {
+    setIsOpen({
+      isOpen: true,
+      loading: true,
+      store: storeInfo,
+    })
+
+    const validation = validateInputs({
+      ...storeInfo,
+    })
+
+    if (!validation) {
+      setError(true)
+
+      return false
+    }
+
+    setError(false)
+
+    return true
+  }
+
+  const handleAddStore = () => {
+    const validated = validate()
+
+    if (validated) {
```

Comment:
You could potentially call validate() directly in the condition, instead of storing the returned value.

```js
const validate = () => {
  return true
}

if (validate()) {
console.log('valid!')
}
```